app/controllers/application_controller.rb

```diff
@@ -6,16 +6,63 @@ def log_in(user)
   end
 
   def log_out
+    forget(current_user)
     session.delete(:user_id)
     @current_user=nil
   end
 
+  def forget(user)
+    user.forget
+    cookies.delete(:user_id)
+    cookies.delete(:remember_token)
+  end 
+
+
+  helper_method :current_user
+  def current_user
+    if (user_id = session[:user_id])
+      @current_user ||= User.find_by(id: user_id)
+    elsif (user_id = cookies.signed[:user_id])
+      user = User.find_by(id: user_id)
+      if user && user.authenticated?(:remember, cookies[:remember_token])
+        log_in user
+        @current_user = user
+      end 
```

Comment:
this block here looks like it's handling the logging in of a user based on the signed cookie. Is there value in storing the user_id in the session if you already have it in the cookie? And if not, could this method be something more simple like:
```
def current_user
  user_id = session[:user_id] || cookies.signed[:user_id]
  @current_user ||= User.find_by(id: user_id) if user_id
end
```